src/Transport/Sending/MessageDispatcher.cs

```diff
@@ -151,7 +151,11 @@ async Task DispatchBatchForDestination(string destination, ServiceBusClient clie
                 else
                 {
                     peekedMessage.ApplicationProperties.TryGetValue(Headers.MessageId, out var messageId);
-                    throw new ServiceBusException($"Message #{messageCount - messagesToSend.Count} with message id '{messageId ?? peekedMessage.MessageId}' is too large and cannot be sent.", ServiceBusFailureReason.MessageSizeExceeded);
+                    var message =
+                        @$"Unable to add the message '#{messageCount - messagesToSend.Count}' with message id '{messageId ?? peekedMessage.MessageId}' to the the batch '#{batchCount}'.
```

Comment:
Well it depends. For example someone could be setting the ID in the code via the options and then when the operation is retried the same ID would be used. For the transactional session, we are also planning to add the possibility to retry stuff and since the operations are already written into the outbox table this ID would give you an indicator on what to look for in the outbox table.

Reply:
Setting their own ID is a marginal use save as the vast majority won't do it. A transactional session, on the other hand, is a more realistic scenario where I can see someone receiving an HTTP request, packaging it just to find out the outgoing message is just too big.